example/m5_competition.py

```diff
@@ -0,0 +1,311 @@
+    os.path.join(work_directory, "raw_calendar.png")
+)
+sell_prices_data.plot(**plot_options).savefig(
+    os.path.join(work_directory, "raw_sell_prices.png")
+)
+
+# Convert time series to tabular dataset
+print("Convert time series to tabular dataset")
+print("======================================")
+
+# We define the computation graph
+sales = sales_data.schema()
+calendar = calendar_data.schema()
+sell_prices = sell_prices_data.schema()
+
+augmented_sales = tp.glue(
+    # Moving average of sales
+    tp.prefix("sma_7.", tp.simple_moving_average(sales, tp.duration.days(7))),
+    tp.prefix("sma_28.", tp.simple_moving_average(sales, tp.duration.days(28))),
+    # Sum of sales
+    tp.prefix("sum_7.", tp.moving_sum(sales, tp.duration.days(7))),
+    tp.prefix("sum_28.", tp.moving_sum(sales, tp.duration.days(28))),
+)
+
+# Lagged sales
+lagged_sales = []
+for lag in [1, 2]:
+    lagged_sales.append(
+        tp.sample(
+            tp.prefix(f"lag_{lag}.", tp.lag(sales, tp.duration.days(lag))),
+            sales,
+        )
+    )
+lagged_sales = tp.glue(*lagged_sales)
+
+
+calendar_events = tp.glue(
+    tp.calendar_day_of_week(sales),
+    tp.calendar_day_of_month(sales),
+    tp.calendar_month(sales),
+)
+
+label_sales = []
+for lag in [1, 2, 3]:
+    label_sales.append(
+        tp.sample(
+            tp.prefix(f"leak_{lag}.", tp.leak(sales, tp.duration.days(lag))),
+            sales,
+        )
+    )
+label_sales = tp.glue(*label_sales)
+
+# Sum of the sales, sampled each day, per departement.
+sales_per_dept = tp.drop_index(sales, "item_id")
+sampling_once_per_day = tp.unique_timestamps(sales_per_dept["item_id"])
+sum_dayly_sales_per_dept = tp.prefix(
+    "per_dept.sum_28_",
+    tp.moving_sum(
+        sales_per_dept["sales"],
+        tp.duration.days(28),
+        sampling_once_per_day,
+    ),
+)
+
+# For each item, add the sum of departement sales for this specific item.
+sales_aggregated_item_level = tp.sample(
+    tp.propagate(sum_dayly_sales_per_dept, sales), sales
+)
+
+# TODO: Last calendar events (need since_last, filter)
+# TODO: Skip early time (start + a few days)
+# TODO: Skip before first sales (need filter)
+# TODO: Comparators + boolean operators
+
+tabular_dataset = tp.glue(
+    sales,  # Raw sales
+    sales_aggregated_item_level,
+    lagged_sales,
+    label_sales,
+    augmented_sales,
+    calendar_events,
+)
+
+# We run the computation
+tabular_dataset_data = tp.evaluate(
+    tabular_dataset,
+    {
+        sales: sales_data,
+        calendar: calendar_data,
+        sell_prices: sell_prices_data,
+    },
+)
+
+# Plot results
+print("Plot results")
+print("============")
+
+tabular_dataset_data.plot(**plot_options).savefig(
+    os.path.join(work_directory, "tabular_dataset.png")
+)
+
+# Export to csv file
+print("Export to csv file")
+print("==================")
+
+tabular_dataset_data.to_dataframe().to_csv(
+    os.path.join(work_directory, "tabular_dataset.csv"), index=False
+)

```

Comment:
nit: print a message saying that plots and csvs were saved to `work_directory` at the end.

Reply:
Ack